pkg/parse/run.go

```diff
@@ -280,63 +301,98 @@ func read(ctx context.Context, p Parser, trigger string, state *reconcilerState,
 	return status.Append(sourceStatus.errs, setSourceStatusErr)
 }
 
-// readFromSource reads the source or hydrated configs, checks whether the sourceState in
-// the cache is up-to-date. If the cache is not up-to-date, reads all the source or hydrated files.
-// readFromSource returns the rendering status and source status.
-func readFromSource(ctx context.Context, p Parser, trigger string, state *reconcilerState, sourceState sourceState) (renderingStatus, sourceStatus) {
-	opts := p.options()
-	start := time.Now()
-
-	hydrationStatus := renderingStatus{
-		commit: sourceState.commit,
-	}
-	sourceStatus := sourceStatus{
-		commit: sourceState.commit,
-	}
-
+// readHydrationStatus reads from the file path which the hydration-controller
+// container writes to. It checks if the hydrated files are ready and returns
+// a renderingStatus.
+func readHydrationStatus(p Parser, srcState sourceState, hydrationStatus renderingStatus) (sourceState, renderingStatus) {
```

Comment:
This function assumes the caller only calls it when hydration controller is running. We should make it clear or move the options.renderingEnabled check logic inside (because someone else might want to use this function later and they should be aware). Also, this function doesn't just read hydrationStatus (it reads files and updates hydrationStatus, so suggest renaming the function.

Reply:
Fair point, refactored this to push the options check into the function and renamed to `parseHydrationState`. I think `parseHydrationState` captures the spirit of the function more precisely than `readHydrationStatus` - note the actual hydrated files are still read by `readConfigFiles`.

I'm a bit concerned that moving the conditional a level down might impact readability of `readFromSource`, but I think it's a fair middle ground.

Reply:
The refactor looks good